aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1107,6 +1116,24 @@ module aptos_framework::stake {
         assert!(exists<StakePool>(pool_address), error::invalid_argument(ESTAKE_POOL_DOES_NOT_EXIST));
     }
 
+    fun validate_network_and_fullnode_addresses(network_addresses: vector<u8>, fullnode_addresses: vector<u8>) {
+        // Fullnode addresses are optional but need to be valid if present.
+        if (vector::length(&fullnode_addresses) > 0) {
+            assert!(
+                validate_network_addresses(network_addresses),
+                error::invalid_argument(EINVALID_NETWORK_ADDRESSES),
+            );
+        };
+
+        // Network addresses are optional but need to be valid if present.
+        if (vector::length(&network_addresses) > 0) {
+            assert!(
+                validate_network_addresses(fullnode_addresses),
+                error::invalid_argument(EINVALID_FULLNODE_ADDRESSES),
```

Comment:
You've got the error codes backwards here. The comment and the check is for network addresses, but the error is for fullnodes.

Reply:
Ah silly me. Thanks for the catch. Fixed now.